lib/mailer.js

```diff
@@ -13,30 +13,46 @@ var Mailer = function () {
   this.user = configuration.getConfig('smtpUsername');
   this.pass = configuration.getConfig('smtpPassword');
   this.from = configuration.getConfig('fromAddress');
+  this.connectionUrl = configuration.getConfig('smtpConnectionUrl');
+
+  this.useSmtpUrl = this.connectionUrl.startsWith('smtp');
 
   // check that required settings exist
   this.validateSettings = function() {
     var errors = [];
 
-    if(this.service === '') errors.push('smtpService');
-    if(this.user === '') errors.push('smtpUsername');
-    if(this.pass === '') errors.push('smtpPassword');
-    if(this.from === '') errors.push('fromAddress');
+    if (this.service === '' && !this.useSmtpUrl) {
+      errors.push('smtpService as a well-known Service or a smtpConnectionUrl is required');
+    } else {
+      if (!this.useSmtpUrl) {
+        if (this.service === '') errors.push('smtpService as a well-known Service');
```

Comment:
@lc-thomasberger I don't see how the second check will ever execute, if both conditions to line 24 are satisfied, it won't ever go into the else.